src/main/java/org/openrewrite/java/spring/data/UseTlsJdbcConnectionString.java

```diff
@@ -79,42 +84,165 @@ protected TreeVisitor<?, ExecutionContext> getVisitor() {
             }
         }
         final String validatedAttribute = attr;
-        return new YamlIsoVisitor<ExecutionContext>() {
-            final JsonPathMatcher jdbcUrl = new JsonPathMatcher("$.spring.datasource.url");
+        return new TreeVisitor<Tree, ExecutionContext>() {
+            @Override
+            public boolean isAcceptable(SourceFile sourceFile, ExecutionContext executionContext) {
+                return sourceFile instanceof Yaml.Documents || sourceFile instanceof Properties.File;
+            }
 
             @Override
-            public Yaml.Mapping.Entry visitMappingEntry(Yaml.Mapping.Entry entry, ExecutionContext ctx) {
-                Yaml.Mapping.Entry e = super.visitMappingEntry(entry, ctx);
-
-                if (jdbcUrl.matches(getCursor())) {
-                    String connectionString = ((Yaml.Scalar) e.getValue()).getValue();
-                    try {
-                        URI jdbcUrl = URI.create(connectionString);
-                        if(oldPort != null && !jdbcUrl.getSchemeSpecificPart().contains(":" + oldPort + "/")) {
-                            return e;
-                        }
-                        URI updatedJdbcUrl = jdbcUrl;
-                        if (port != null && !jdbcUrl.getSchemeSpecificPart().contains(":" + port + "/")) {
-                            updatedJdbcUrl = new URI(jdbcUrl.getScheme(), jdbcUrl.getSchemeSpecificPart()
-                                    .replaceFirst(":\\d+/", ":" + port + "/"), jdbcUrl.getFragment());
-                        }
-                        if (validatedAttribute != null && !jdbcUrl.getSchemeSpecificPart().contains(validatedAttribute)) {
-                            updatedJdbcUrl = new URI(updatedJdbcUrl.getScheme(),
-                                    updatedJdbcUrl.getSchemeSpecificPart() +
-                                    (updatedJdbcUrl.getSchemeSpecificPart().endsWith(";") ? "" : ":") +
-                                            validatedAttribute,
-                                    updatedJdbcUrl.getFragment());
+            public @Nullable Tree visit(@Nullable Tree tree, ExecutionContext ctx) {
+                if (tree instanceof SourceFile) {
+                    return super.visit(tree, ctx);
+                }
+                return tree;
+            }
+
+            @Override
+            public Tree preVisit(Tree tree, ExecutionContext ctx) {
+                if (tree instanceof Yaml.Documents) {
+                    doAfterVisit(new UseTlsJdbcConnectionStringYaml(oldPort, port, validatedAttribute));
+                } else if (tree instanceof Properties.File) {
+                    doAfterVisit(new UseTlsJdbcConnectionStringProperties(oldPort, port, validatedAttribute));
+                }
+                return tree;
+            }
```

Comment:
Could simplify this a bit
```suggestion
            public @Nullable Tree visit(@Nullable Tree tree, ExecutionContext ctx) {
                if (tree instanceof Yaml.Documents) {
                    return new UseTlsJdbcConnectionStringYaml(oldPort, port, validatedAttribute).visitNonNull(tree, ctx);
                } else if (tree instanceof Properties.File) {
                    return new UseTlsJdbcConnectionStringProperties(oldPort, port, validatedAttribute).visitNonNull(tree, ctx);
                }
                return tree;
            }
```